### ISSUE TITLE
Fails to display docs for a go-gettable (after a human verifies SSH host one time) package.

### ISSUE THREAD

**Issue by shurcooL** [golang/gddo#510]
The Go package with a vanity import path "dmitri.shuralyov.com/kebabcase" can be acquired with `go get`:

``` bash
$ go get -u -d dmitri.shuralyov.com/kebabcase
$ 
```

It also works on gotools.org:

https://gotools.org/dmitri.shuralyov.com/kebabcase

However, it results in a "Not Found" page on godoc.org:

https://godoc.org/dmitri.shuralyov.com/kebabcase

![image](https://user-images.githubusercontent.com/1924134/30312461-a57d3222-9767-11e7-84d8-57519b5d6299.png)

At this time, its go-import meta tag is:

```
$ curl -s https://dmitri.shuralyov.com/kebabcase?go-get=1 | grep go-import
<meta name="go-import" content="dmitri.shuralyov.com/kebabcase git ssh://git@dmitri.shuralyov.com:2022/kebabcase">
```

(Note, the exact `<meta name="go-import" content="...">` setup of this Go package is not final, it may change in the near future. However, I found it odd that godoc.org can't display a go-gettable package, and want to fix this bug first.)

## Investigation

I'm not sure what the cause is. This Go package has some rare and unique properties, and it seems godoc.org doesn't handle some edge case.

I tried to reproduce the issue locally and could not. It worked fine locally:

![image](https://user-images.githubusercontent.com/1924134/30312544-0a696a7a-9768-11e7-9aa5-d762f136672d.png)

(I tried with a slightly older version, with changes on top to remove app engine dependencies. But hopefully that's not relevant.)

Can someone who has access to production logs (/cc @shantuo @stephenmw) look to see what the actual error was? Then I could investigate further, if the problem is in the source code.

**Comment by shantuo** [comment]
From the log:
`notfound: VCS not found dmitri.shuralyov.com/kebabcase`
The command gddo uses:
`git ls-remote --heads --tags https://dmitri.shuralyov.com/kebabcase`

**Comment by shantuo** [comment]
When running locally
```
2017/09/12 11:30:27 git ls-remote --heads --tags ssh://git@dmitri.shuralyov.com:2022/kebabcase
The authenticity of host '[dmitri.shuralyov.com]:2022 ([172.93.50.41]:2022)' can't be established.
RSA key fingerprint is SHA256:KbIJ2+ylBglIPmqaoGLvNlNwjsAFkKWzo9Te1MHr1fE.
Are you sure you want to continue connecting (yes/no)? 2017/09/12 11:30:31 Serving "dmitri.shuralyov.com/kebabcase" as not found after timeout getting doc

**Comment by shurcooL** [comment]
`go get` behavior is to try multiple protocols. From https://golang.org/cmd/go/#hdr-Remote_import_paths:

> When a version control system supports multiple protocols, each is tried in turn when downloading. For example, a Git download tries https://, then git+ssh://.

The only protocol this repository host supports at this time is `ssh://`, one of the supported protocols for git.

It seems the problem is that it stalls because of asking a human to verify if the ssh key fingerprint is valid, times out, and reports the package as not found.

What do you think is the best way to proceed to resolve this issue?

**Comment by shurcooL** [comment]
I ran an experiment on a completely fresh VM with Ubuntu, to see what would happen there.

This is the user experience of trying to `go get -u` this package on a machine that has never tried to connect to that server via SSH:

```
...
Preparing to unpack .../git_1%3a2.11.0-2_amd64.deb ...
Unpacking git (1:2.11.0-2) ...
Setting up git-man (1:2.11.0-2) ...
Setting up liberror-perl (0.17024-1) ...
Processing triggers for man-db (2.7.6.1-2) ...
Setting up git (1:2.11.0-2) ...
ubuntu@ubuntu:~$ go version
go version go1.9 linux/amd64
ubuntu@ubuntu:~$ go get -u -d dmitri.shuralyov.com/kebabcase
The authenticity of host '[dmitri.shuralyov.com]:2022 ([172.93.50.41]:2022)' can't be established.
RSA key fingerprint is SHA256:KbIJ2+ylBglIPmqaoGLvNlNwjsAFkKWzo9Te1MHr1fE.
Are you sure you want to continue connecting (yes/no)? 
```

So it happens with `go get` on a fresh VM that doesn't have that SSH server added to its `known_hosts` file.

When using HTTPS, things work out of the box because of trusted (by default) chains. So, for example, one is able to get an HTTPS certificate from Let's Encrypt for their domain name, and a brand new computer that has never seen that HTTPS server will be able to trust it via Let's Encrypt.

I'm not sure if something equivalent and easily accessible exists for SSH (although I've been pointed to [this](https://www.digitalocean.com/community/tutorials/how-to-create-an-ssh-ca-to-validate-hosts-and-clients-with-ubuntu)).

Here are the possible solutions I see:

- Manually add the missing entry to `known_hosts` for that server. This is bad because it doesn't scale. New servers will manually need to be added with human intervention only.

- Use `StrictHostKeyChecking=no` SSH option. This is bad because of negative security implications.

- Do not support SSH-only because it's not practical. This is bad because it's unfortunate. SSH has nice properties, and I'd rather not have to abandon it (or support additional alternative schemes, such as HTTPS) just so that the documentation of a Go package can be seen via godoc.org.

None of the options are particularly attractive. Any ideas or suggestions would be highly appreciated.

**Comment by shurcooL** [comment]
I've thought about this some more, and come to the following conclusion.

It doesn't seem viable to distribute Go packages via SSH protocol, so I've abandoned that idea for now, and went with HTTPS instead.

I'm going to close this issue. Nothing needs to be done on godoc.org side.

(I've changed the `<meta name="go-import" content="...">` tag to use https:// protocol and https://godoc.org/dmitri.shuralyov.com/kebabcase is working okay now.)
